### DISCUSSION TITLE
Performance of EditObject vs MudDialog

### DISCUSSION THREAD

**Discussion by PmE8HW0KRfqa** [fgilde/MudBlazor.Extensions#72] [General]
@fgilde, first I'd like to thank you for providing this great tool. My query is by no means an attempt to cast any shade on the work you've done. Secondly, my experience with Blazor and related technologies is limited and I wouldn't be surprised if this is the root cause of my observation.

When comparing the EditObject against MudDialog (with databinding), I've noticed the MudDialog pops up noticably quicker than the EditObject dialog. Is EditObject inherently a lot slower, or am I just using it improperly? Is this a problem with my EditObject/DialogOptionsEx configuration?

When I open an EditObject dialog, the following happens:
The page background darkness (this appears to be a ~3 frame animation),
The dialog fades-in (again ~3 frames)

The MudDialog does the same, but it's far faster.

**Comment by fgilde** [comment]
In general, it's not directly comparable. The MudDialog is not intended to generate the content of the dialog based on the model. If you have the UI in a Razor control and want to bind it yourself, you can simply use a standard MudDialog. The extension method `dialogService.EditObject` utilizes the `MudExObjectEditDialog`, which in turn also uses the standard `MudDialog`. The `MudExObjectEditDialog` is designed to generate the content of the dialog based on your model and the MetaConfiguration. As for the speed, it depends on several factors. For one, in a WebAssembly application, it partly depends on the hardware itself, and then, of course, on the complexity of the model and the MetaConfiguration.

I've created a sample for you here that renders a somewhat more complex model. It works quite swiftly for me.

https://try.mudex.org/snippet/muQoOQPAoYGsmwAi
